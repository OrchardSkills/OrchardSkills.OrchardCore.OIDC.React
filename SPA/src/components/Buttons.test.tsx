import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Buttons from './Buttons';

describe(Buttons, () => {
  const mockLogin = jest.fn();
  const mockLogout = jest.fn();

  const component = shallow(
    <Buttons
      login={mockLogin}
      logout={mockLogout}

    />
  );

  it('renders and matches our snapshot', () => {
    const buttonComponent = renderer.create(
      <Buttons
        login={mockLogin}
        logout={mockLogout}
      />
    );
    const tree = buttonComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the image and h1', () => {
    expect(component.find('button')).toHaveLength(2);
  });

  it('try to click on login button', () => {
    component.find('button.btn-login').simulate('click');
    expect(mockLogin).toBeCalled();
  });

  it('try to click on logout button', () => {
    component.find('button.btn-logout').simulate('click');
    expect(mockLogout).toBeCalled();
  });
});
