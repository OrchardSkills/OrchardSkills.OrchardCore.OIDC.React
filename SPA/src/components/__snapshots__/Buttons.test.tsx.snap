// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders and matches our snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12 text-center"
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <button
      className="btn btn-primary btn-login"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Login
    </button>
    <button
      className="btn btn-dark btn-logout"
      onClick={[MockFunction]}
      style={
        Object {
          "margin": "10px",
        }
      }
    >
      Logout
    </button>
  </div>
</div>
`;
